fetch.bs

```diff
@@ -8713,82 +8713,142 @@ relevant to you. The rest of the parameters are used less frequently, often for
+   <ol>
+    <li><p>Let <var>request</var> be a <a for=/>request</a> whose <a for=request>URL</a> is
+    "<code>https://stuff.example.com</code>", and whose <a for=request>client</a> is <a>this</a>'s
+    <a>relevant settings object</a>.
+
+    <li>
+     <p><a for=/>Fetch</a> <var>request</var>, with
+     <a for=fetch><i>processResponseConsumeBody</i></a> set to the following steps given
+     a <a for=/>response</a> <var>response</var> and a null, failure or <a>byte sequence</a>
+     <var>contents</var>:
+
+     <ol>
+      <li><p>If <var>contents</var> is null or failure, then present an error to the user.
+
+      <li><p>Otherwise, parse <var>contents</var> considering the metadata from <var>response</var>,
+      and perform your own operations on it.
+     </ol>
+    </li>
+  </div>
+ </dd>
 
- <dt><a for=fetch><i>processResponseConsumeBody</i></a>
+ <dt>Headers first, then Chunk-by-chunk
  <dd>
-  <p>Takes an algorithm that will be passed a <a for=/>response</a> and null, failure, or a
-  <a>byte sequence</a>. This is useful for standards that wish to operate on the entire
-  <a for=/>response</a>'s <a for=response>body</a>, of which the result of reading it is supplied as
-  second argument. The second argument's values have the following meaning:
-
-  <dl>
-   <dt>null
-   <dd>The <a for=/>response</a>'s <a for=response>body</a> is null, due to the response being a
-   <a for=/>network error</a> or having a <a>null body status</a>.
-
-   <dt>failure
-   <dd>Attempting to <a for=body>fully read</a> the contents of the <a for=/>response</a>'s
-   <a for=response>body</a> failed, e.g., due to an I/O error.
-
-   <dt>a <a>byte sequence</a>
-   <dd><a for=body>Fully reading</a> the contents of the <a for=/>response</a>'s
-   <a for=response>body</a> succeeded.
-  </dl>
-
-  <p class=warning>A standard that uses this argument cannot operate on <a for=/>response</a>'s
-  <a for=response>body</a> itself as providing this argument will cause it to be read and it can be
-  read only once.
+  <p>In some cases, for example when playing video or progressively loading images, clients might
+  want to stream the response, and process it one chunk at a time. The <a for=/>response</a> is
+  handed over to the Fetch caller once the headers are processed, and the caller
+  continues from there.
+
+  <p>To process a <a for=/>response</a> chunk-by-chunk, pass an algorithm to the
+  <a for=fetch><i>processResponse</i></a> argument of <a for/>fetch</a>. The given
+  algorithm is passed a <a for=/>response</a> when the response's headers have been
+  received and is responsible for reading the <a for=/>response</a>'s
+  <a for=response>body</a>'s <a for=body>stream</a> in order to download the rest
+  of the response. For convenience, you may also pass an algorithm to the
+  <a for=fetch><i>processResponseEndOfBody</i></a> argument, which is called once you have finished
+  obtaining the response and its <a for=response>body</a>. Note that unlike
```

Comment:
Maybe "fully reading the response's body"? I don't know what it means to "obtain the response and its body".

You keep using "obtain" below as well... I think you read bodies, you don't obtain them.

I see @annevk seems to be advocating for "obtain" though, so maybe I'm off-base?

Reply:
Yea I also wasn't so much into "obtain". @annevk are you ok with "fully reading"?

Reply:
Fully reading seems fine.